### PR TITLE
Add Loading Spinners to Components in `home.py`

### LINKED ISSUE #134: User and Sign Ups Bug on Overview
There is a **bug on the Overview/Home page** where certain components do not load properly:
- **# Active Users**
- **# Users**
- **Sign Ups Trend**

## Steps to Reproduce

1. Navigate to the data page.
2. Change the date on the top right to a future date (e.g., November 2023).
3. Return to the Overview/Home page.
4. Observe that the following components fail to load:
   - **# Active Users**
   - **# Users**
   - **Sign Ups Trend**
   
However, other components such as **# Confirmed Trips** and **Trips Trend** continue to function normally.

## Expected Behavior

All components on the Overview/Home page should load and update properly, regardless of the date changes made on the data page.

## Actual Behavior

After changing the date on the data page and returning to the Overview/Home page, the **# Active Users**, **# Users**, and **Sign Ups Trend** components fail to load/update.

### PR DESCRIPTION
**Pull request by TeachMeTW**
## Summary

This pull request introduces skeleton loading screens for the components in `home.py`, providing better user feedback during data fetching and updates. The skeleton loading states will appear while waiting for all components to finish loading, ensuring a smooth transition and improved user experience.

## Changes

- Added `dcc.Loading` components with skeleton loading states to:
  - User, Active Users, and Trips cards.
  - Sign-up and Trips trend graphs.
  
- The skeleton loading will remain active until all components are fully loaded, at which point the actual data will be displayed.

- Adjusted callbacks to ensure they trigger the skeleton loaders when necessary, providing real-time feedback during data fetching.

## Impact

- **Improved User Experience**: The addition of skeleton loaders gives users a smoother visual feedback during operations, making the app feel more responsive and polished by preventing partial renders.

## Demo:

https://github.com/user-attachments/assets/27351871-b079-4118-a32e-4ea27675522f

  
Resolves #134

### CODE REVIEW

**Comment by TeachMeTW** [comment]
CURRENTLY WIP:
Skeleton Loading/Differentiation between component and page

**Comment by TeachMeTW** [comment]
Accidentally commited to wrong branch; tried to fix

**Comment by TeachMeTW** [comment]
@JGreenlee @shankari 

I've implemented skeleton loading for the app, but I’m unsure about one detail. Right now, some components load faster than others, creating a staggered loading experience. I’m considering whether it would be better to have all components wait for the slowest one to finish loading before displaying everything together.

Visually, it seems more cohesive to have everything appear at once, but I’m concerned this might negatively impact perceived performance since faster components are delayed.

Would love to hear your thoughts on which approach is better from a UX/design perspective:

Staggered loading (as it is now)
Unified loading (wait for all components to finish before displaying)


(the video is not up to date but skeleton is implemented)

File: app_sidebar_collapsible.py [hunk 1]
```diff
@@ -385,4 +414,4 @@ def display_page(search):
         SESSION_COOKIE_HTTPONLY=True
     )
     logging.debug("after override, current server config = %s" % server.config)
-    app.run_server(debug=envDebug, host='0.0.0.0', port=envPort)
+    app.run_server(debug=envDebug, host='0.0.0.0', port=envPort)
```
Comment (JGreenlee) [inline app_sidebar_collapsible.py:417]: whitespace change

File: pages/home.py [hunk 2]
```diff
@@ -1,12 +1,7 @@
-"""
-Note that the callback will trigger even if prevent_initial_call=True. This is because dcc.Location must
-be in app.py.  Since the dcc.Location component is not in the layout when navigating to this page, it triggers the callback.
-The workaround is to check if the input value is None.
-
-"""
 from uuid import UUID
-from dash import dcc, html, Input, Output, callback, register_page
+from dash import dcc, html, Input, Output, callback, register_page, no_update
```
Comment (JGreenlee) [inline pages/home.py:2]: It's not clear to me what `no_update` does or why it's added. Can you add a comment in the code with an explanation (or a link to Dash documentation that explains it)?

File: pages/home.py [hunk 3]
```diff
@@ -117,73 +132,104 @@ def generate_card(title_text, body_text, icon):
...
+        Input('store-trips', 'data'),
+        Input('date-picker', 'start_date'),
+        Input('date-picker', 'end_date'),
+        Input('url', 'pathname'),
+        Input('home-page-load', 'children')
+    ]
 )
-def update_card_active_users(store_uuids):
-    uuid_df = pd.DataFrame(store_uuids.get('data'))
+def update_all_components(store_uuids, store_trips, start_date, end_date, pathname, _):
```
Comment (JGreenlee) [inline pages/home.py:164]: I am seeing that we used to have multiple, smaller callbacks for each card. Now we have one big callback that handles everything– which is worse from a code tidiness standpoint.
Can you explain the rationale behind this?

**Review by JGreenlee — CHANGES_REQUESTED** [review]

**Pull request closed without merging**